src/app/files.js

```diff
@@ -0,0 +1,119 @@
+'use strict'
+
+var EventManager = require('../lib/eventManager')
+
+function Files (storage) {
+  var event = new EventManager()
+  this.event = event
+  var readonly = {}
+
+  this.exists = function (path) {
+    // NOTE: ignore the config file
+    if (path === '.browser-solidity.json') {
+      return false
+    }
+
+    if (this.isReadOnly(path)) {
+      return true
+    } else {
+      return storage.exists(path)
+    }
+  }
+
+  this.get = function (path) {
+    // NOTE: ignore the config file
+    if (path === '.browser-solidity.json') {
+      return null
+    }
+
+    if (this.isReadOnly(path)) {
+      return readonly[path]
+    } else {
+      return storage.get(path)
+    }
+  }
+
+  this.set = function (path, content) {
+    // NOTE: ignore the config file
+    if (path === '.browser-solidity.json') {
+      return false
+    }
+
+    if (!this.isReadOnly(path)) {
+      var exists = storage.exists(path)
+      if (!storage.set(path, content)) {
+        return false
+      }
+      if (!exists) {
+        event.trigger('fileAdded', [path])
+      } else {
+        event.trigger('fileChanged', [path])
+      }
+      return true
+    }
+
+    return false
+  }
+
+  this.addReadOnly = function (path, content) {
+    if (!storage.exists(path)) {
+      readonly[path] = content
+      event.trigger('fileAdded', [path])
```

Comment:
We could supply the path and the readonly property, so that the UI does not have to ask again.